internal/application/manual_game_service.go

```diff
@@ -445,6 +445,14 @@ func (s *ManualGameService) removeCardFromDeck(card domain.Card) error {
 		return fmt.Errorf("no active round/deck")
 	}
 	deck := s.Game.CurrentRound.Deck
+
+	// For number cards, check RemainingCounts first for efficiency
+	if card.Type == domain.CardTypeNumber {
+		if count, exists := deck.RemainingCounts[card.Value]; !exists || count <= 0 {
+			return fmt.Errorf("card not found in deck (already drawn?)")
+		}
+	}
```

Comment:
I found the case to reproduce the bug.

1. Set the number of player be 2.
2. Type 4 for 4 times(i.e., each player drew 5 for two times and busted)
3. At next round, type 4.

Expected: return error (`fmt.Errorf("card not found in deck (already drawn?)")`)
Result: return nil, i.e., the turn passes to next player.

The above problem is not solved in this change. And your no-effect changes suggest that there is a problem on handling Deck with continuing game.